# Exploratory Data Analysis


In [1]:
import pandas as pd


In [2]:
!ls zippedData

bom.movie_gross.csv.gz
imdb.name.basics.csv.gz
imdb.title.akas.csv.gz
imdb.title.basics.csv.gz
imdb.title.crew.csv.gz
imdb.title.principals.csv.gz
imdb.title.ratings.csv.gz
rt.movie_info.tsv.gz
rt.reviews.tsv.gz
tmdb.movies.csv.gz
tn.movie_budgets.csv.gz


In [3]:
bom = pd.read_csv('zippedData/bom.movie_gross.csv.gz')
num = pd.read_csv('zippedData/tn.movie_budgets.csv.gz')

## Box Office Mojo



This data is relevant because it tells us both the domestic and foreign gross incomes for each movie. <br>
We need to know how much income previous movies generated in order to advise microsoft on what kind <br> 
of movies to focus on.

We know the source is credible because it is owned by world reknown IMDb.  <br>





In [4]:
bom.head(20)


title studio  domestic_gross  \
0                                         Toy Story 3     BV     415000000.0   
1                          Alice in Wonderland (2010)     BV     334200000.0   
2         Harry Potter and the Deathly Hallows Part 1     WB     296000000.0   
3                                           Inception     WB     292600000.0   
4                                 Shrek Forever After   P/DW     238700000.0   
5                          The Twilight Saga: Eclipse   Sum.     300500000.0   
6                                          Iron Man 2   Par.     312400000.0   
7                                             Tangled     BV     200800000.0   
8                                       Despicable Me   Uni.     251500000.0   
9                            How to Train Your Dragon   P/DW     217600000.0   
10                         Clash of the Titans (2010)     WB     163200000.0   
11  The Chronicles of Narnia: The Voyage of the Da...    Fox     104400000.0   
12                                  The King's Speech  Wein.     135500000.0   
13                                        Tron Legacy     BV     172100000.0   
14                                     The Karate Kid   Sony     176600000.0   
15                Prince of Persia: The Sands of Time     BV      90800000.0   
16                                         Black Swan   FoxS     107000000.0   
17                                           Megamind   P/DW     148400000.0   
18                                         Robin Hood   Uni.     105300000.0   
19                                 The Last Airbender   Par.     131800000.0   

   foreign_gross  year  
0      652000000  2010  
1      691300000  2010  
2      664300000  2010  
3      535700000  2010  
4      513900000  2010  
5      398000000  2010  
6      311500000  2010  
7      391000000  2010  
8      291600000  2010  
9      277300000  2010  
10     330000000  2010  
11     311300000  2010  
12     275400000  2010  
13     228000000  2010  
14     182500000  2010  
15     245600000  2010  
16     222400000  2010  
17     173500000  2010  
18     216400000  2010  
19     187900000  2010

In [14]:
bom.tail(10)

title      studio  domestic_gross foreign_gross  \
3377               Let Yourself Go        Men.         12200.0           NaN   
3378                 Hannah (2018)         PDF         11700.0           NaN   
3379                      Souvenir      Strand         11400.0           NaN   
3380  Furious (Legend of Kolovrat)     CARUSEL         10000.0           NaN   
3381           Beauty and the Dogs       Osci.          8900.0           NaN   
3382                     The Quake       Magn.          6200.0           NaN   
3383   Edward II (2018 re-release)          FM          4800.0           NaN   
3384                      El Pacto        Sony          2500.0           NaN   
3385                      The Swan  Synergetic          2400.0           NaN   
3386             An Actor Prepares       Grav.          1700.0           NaN   

      year  
3377  2018  
3378  2018  
3379  2018  
3380  2018  
3381  2018  
3382  2018  
3383  2018  
3384  2018  
3385  2018  
3386  2018

In [6]:
bom.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3387 entries, 0 to 3386
Data columns (total 5 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   title           3387 non-null   object 
 1   studio          3382 non-null   object 
 2   domestic_gross  3359 non-null   float64
 3   foreign_gross   2037 non-null   object 
 4   year            3387 non-null   int64  
dtypes: float64(1), int64(1), object(3)
memory usage: 132.4+ KB


In [7]:
bom.isnull().sum()

title                0
studio               5
domestic_gross      28
foreign_gross     1350
year                 0
dtype: int64

##### Our dataset has 3387 records and 5 categories.  We have 3 columns with missing data. <br>
The 'Studio' column is only missing 5 entries, and our 'domestic_gross' column is missing 28 entries. <br>
These numbers a miniscule when compared to the whole dataset, but our 'foreign_gross' column is missing <br>
almost a 1/3rd of the entries.  This data set is ordered by the date column.  It contains movie data between  <br>
the years 2010 and 2018.

As we can see, 'foreign_gross' is currently a string, so I will have to convert that to an integer. 

In [8]:
bom.describe()

domestic_gross         year
count    3.359000e+03  3387.000000
mean     2.874585e+07  2013.958075
std      6.698250e+07     2.478141
min      1.000000e+02  2010.000000
25%      1.200000e+05  2012.000000
50%      1.400000e+06  2014.000000
75%      2.790000e+07  2016.000000
max      9.367000e+08  2018.000000

###### Data Limitations

I cant see how much the studio spent to make each movie in this dataset.  I have no way to see if the they are making a profit on each movie.  I also do not see how the studio column helps me advise Microsoft.  Microsoft is using their own studio so that column doesn't really help in this case.

Also this dataset does not contain foreign gross for the more recent movies.  So it will be difficult to determine worldwide income

## The Numbers 

The numbers is a website owned by Nash Information Services. They are seen as the premier source for Movie Data and research data.

###### Why is this information relevant

On top of the gross income that each movie generated, this table also tells us how much each movie costs to make.  If we subtract the production_budget from the worldwide gross we will be able to see how much profit each movie brought in.

In [15]:
num.head(20)

id  release_date                                        movie  \
0    1  Dec 18, 2009                                       Avatar   
1    2  May 20, 2011  Pirates of the Caribbean: On Stranger Tides   
2    3   Jun 7, 2019                                 Dark Phoenix   
3    4   May 1, 2015                      Avengers: Age of Ultron   
4    5  Dec 15, 2017            Star Wars Ep. VIII: The Last Jedi   
5    6  Dec 18, 2015         Star Wars Ep. VII: The Force Awakens   
6    7  Apr 27, 2018                       Avengers: Infinity War   
7    8  May 24, 2007   Pirates of the Caribbean: At World’s End   
8    9  Nov 17, 2017                               Justice League   
9   10   Nov 6, 2015                                      Spectre   
10  11  Jul 20, 2012                        The Dark Knight Rises   
11  12  May 25, 2018                      Solo: A Star Wars Story   
12  13   Jul 2, 2013                              The Lone Ranger   
13  14   Mar 9, 2012                                  John Carter   
14  15  Nov 24, 2010                                      Tangled   
15  16   May 4, 2007                                 Spider-Man 3   
16  17   May 6, 2016                   Captain America: Civil War   
17  18  Mar 25, 2016           Batman v Superman: Dawn of Justice   
18  19  Dec 14, 2012            The Hobbit: An Unexpected Journey   
19  20  Jul 15, 2009       Harry Potter and the Half-Blood Prince   

   production_budget domestic_gross worldwide_gross  
0       $425,000,000   $760,507,625  $2,776,345,279  
1       $410,600,000   $241,063,875  $1,045,663,875  
2       $350,000,000    $42,762,350    $149,762,350  
3       $330,600,000   $459,005,868  $1,403,013,963  
4       $317,000,000   $620,181,382  $1,316,721,747  
5       $306,000,000   $936,662,225  $2,053,311,220  
6       $300,000,000   $678,815,482  $2,048,134,200  
7       $300,000,000   $309,420,425    $963,420,425  
8       $300,000,000   $229,024,295    $655,945,209  
9       $300,000,000   $200,074,175    $879,620,923  
10      $275,000,000   $448,139,099  $1,084,439,099  
11      $275,000,000   $213,767,512    $393,151,347  
12      $275,000,000    $89,302,115    $260,002,115  
13      $275,000,000    $73,058,679    $282,778,100  
14      $260,000,000   $200,821,936    $586,477,240  
15      $258,000,000   $336,530,303    $894,860,230  
16      $250,000,000   $408,084,349  $1,140,069,413  
17      $250,000,000   $330,360,194    $867,500,281  
18      $250,000,000   $303,003,568  $1,017,003,568  
19      $250,000,000   $302,089,278    $935,213,767

In [10]:
num.tail(13)

id  release_date                            movie production_budget  \
5769  70   Apr 1, 1996                             Bang           $10,000   
5770  71  Aug 14, 2008  The Rise and Fall of Miss Thang           $10,000   
5771  72  May 19, 2015                 Family Motocross           $10,000   
5772  73  Jan 13, 2012                        Newlyweds            $9,000   
5773  74  Feb 26, 1993                      El Mariachi            $7,000   
5774  75   Oct 8, 2004                           Primer            $7,000   
5775  76  May 26, 2006                           Cavite            $7,000   
5776  77  Dec 31, 2004                  The Mongol King            $7,000   
5777  78  Dec 31, 2018                           Red 11            $7,000   
5778  79   Apr 2, 1999                        Following            $6,000   
5779  80  Jul 13, 2005    Return to the Land of Wonders            $5,000   
5780  81  Sep 29, 2015             A Plague So Pleasant            $1,400   
5781  82   Aug 5, 2005                My Date With Drew            $1,100   

     domestic_gross worldwide_gross  
5769           $527            $527  
5770           $401            $401  
5771             $0              $0  
5772         $4,584          $4,584  
5773     $2,040,920      $2,041,928  
5774       $424,760        $841,926  
5775        $70,071         $71,644  
5776           $900            $900  
5777             $0              $0  
5778        $48,482        $240,495  
5779         $1,338          $1,338  
5780             $0              $0  
5781       $181,041        $181,041

In [12]:
num['domestic_gross'].value_counts()

$0              548
$8,000,000        9
$7,000,000        7
$2,000,000        7
$10,000,000       6
               ... 
$33,618,855       1
$79,957,634       1
$86,300,000       1
$102,272,145      1
$17,472           1
Name: domestic_gross, Length: 5164, dtype: int64

In [28]:
num.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5782 entries, 0 to 5781
Data columns (total 6 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   id                 5782 non-null   int64 
 1   release_date       5782 non-null   object
 2   movie              5782 non-null   object
 3   production_budget  5782 non-null   object
 4   domestic_gross     5782 non-null   object
 5   worldwide_gross    5782 non-null   object
dtypes: int64(1), object(5)
memory usage: 271.2+ KB


###### Dataset features

This dataset has 5782 records and 7 columns including the index. The good thing about this dataset is that it has no missing values.  To use this dataset I will have to convert the gross income and budget columns to integers.  I will also have to strip those columns of the '$' signs and commas before I can perform statistical functions on them.  I also noticed that this dataset is ordered by the 'production_budget' column in descending order.

###### Dataset limitations

There is no 'genre' column in this dataset.  So I cant get statistical data on certain genres of movies, and see how they relate to profit.

# Data Cleaning




<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3387 entries, 0 to 3386
Data columns (total 5 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   title           3387 non-null   object 
 1   studio          3382 non-null   object 
 2   domestic_gross  3359 non-null   float64
 3   foreign_gross   2037 non-null   object 
 4   year            3387 non-null   int64  
dtypes: float64(1), int64(1), object(3)
memory usage: 132.4+ KB


In [21]:
# I will work on a copy of the dataset so that I do not alter the original,
# until I am sure that I want to make these changes permanent.

ncopy = num.copy()

id  release_date                                        movie  \
0   1  Dec 18, 2009                                       Avatar   
1   2  May 20, 2011  Pirates of the Caribbean: On Stranger Tides   
2   3   Jun 7, 2019                                 Dark Phoenix   
3   4   May 1, 2015                      Avengers: Age of Ultron   
4   5  Dec 15, 2017            Star Wars Ep. VIII: The Last Jedi   

  production_budget domestic_gross worldwide_gross  
0      $425,000,000   $760,507,625  $2,776,345,279  
1      $410,600,000   $241,063,875  $1,045,663,875  
2      $350,000,000    $42,762,350    $149,762,350  
3      $330,600,000   $459,005,868  $1,403,013,963  
4      $317,000,000   $620,181,382  $1,316,721,747

In [47]:
bcopy.isnull().sum()

title             0
studio            5
domestic_gross    0
foreign_gross     0
year              0
dtype: int64

This just a copy so I want to convert the columns to integers so that I can gain statistical knowledge on the data.  I will be more strategic about filling in missing values on the real dataset.